fendermint/vm/interpreter/src/fvm/activity/mod.rs

```diff
@@ -25,6 +27,35 @@ pub trait ValidatorActivityTracker {
     fn commit_activity(&mut self) -> anyhow::Result<FullActivity>;
 }
 
+impl TryFrom<fendermint_actor_activity_tracker::types::FullActivityRollup> for FullActivity {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: fendermint_actor_activity_tracker::types::FullActivityRollup,
+    ) -> Result<Self, Self::Error> {
+        let f = FullActivityRollup {
+            consensus: FullSummary {
+                stats: AggregatedStats {
+                    total_active_validators: value.consensus.stats.total_active_validators,
+                    total_num_blocks_committed: value.consensus.stats.total_num_blocks_committed,
+                },
+                data: value
+                    .consensus
+                    .data
+                    .into_iter()
+                    .map(|(addr, data)| {
+                        Ok(ValidatorData {
+                            validator: payload_to_evm_address(addr.payload())?,
+                            blocks_committed: data.blocks_committed,
+                        })
+                    })
+                    .collect::<anyhow::Result<Vec<_>>>()?,
+            },
+        };
```

Comment:
I kinda hate this deeply nested code. I would much prefer you to use an expression here, create the stats object, then the data object, then return the FullActivityRollup.